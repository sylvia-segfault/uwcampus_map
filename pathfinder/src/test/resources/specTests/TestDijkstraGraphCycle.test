# Find a path with Dijkstra's algorithm on
# a big graph that contains cycles

CreateGraph graph
AddNode graph A
AddNode graph B
AddNode graph C
AddNode graph D
AddNode graph E
AddNode graph F
AddNode graph G
AddNode graph H
AddNode graph I
AddEdge graph C D 1.5
AddEdge graph C E 3.0
AddEdge graph C H 5.0
AddEdge graph B C 2.5
AddEdge graph B A 3.0
AddEdge graph A H 2.0
AddEdge graph H G 6.0
AddEdge graph G I 7.0
AddEdge graph I D 8.0
AddEdge graph D F 1.5
AddEdge graph E F 1.0
FindPath graph B F