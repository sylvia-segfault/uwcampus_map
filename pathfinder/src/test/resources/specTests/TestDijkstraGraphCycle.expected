# Find a path with Dijkstra's algorithm on
# a big graph that contains cycles

created graph graph
added node A to graph
added node B to graph
added node C to graph
added node D to graph
added node E to graph
added node F to graph
added node G to graph
added node H to graph
added node I to graph
added edge 1.500 from C to D in graph
added edge 3.000 from C to E in graph
added edge 5.000 from C to H in graph
added edge 2.500 from B to C in graph
added edge 3.000 from B to A in graph
added edge 2.000 from A to H in graph
added edge 6.000 from H to G in graph
added edge 7.000 from G to I in graph
added edge 8.000 from I to D in graph
added edge 1.500 from D to F in graph
added edge 1.000 from E to F in graph
path from B to F:
B to C with weight 2.500
C to D with weight 1.500
D to F with weight 1.500
total cost: 5.500